services/provider/server/server.go

```diff
@@ -358,7 +358,9 @@ func (s *OCSProviderServer) GetDesiredClientState(ctx context.Context, req *pb.G
 			}
 			kubeResource.GetObjectKind().SetGroupVersionKind(gvk)
 			sanitizeKubeResource(kubeResource)
-			response.KubeResources = append(response.KubeResources, mustMarshal(kubeResource))
+			kubeResourceBytes := mustMarshal(kubeResources)
```

Comment:
```suggestion
			kubeResourceBytes := mustMarshal(kubeResource)
```
Taking a closer look, CI could be failing due to this.